### PR TITLE
Change the way UserProfileModel is initialize

### PR DESCRIPTION
**Pull request by jeremierobert-coveo**
I change the way to initilize the model of the user action as an atempt to simplify the process.

### CODE REVIEW

File: src/components/UserActions/ClickedDocumentList.ts [hunk 1]
```diff
@@ -90,13 +90,18 @@ export class ClickedDocumentList extends Component {
         super(element, ClickedDocumentList.ID, bindings);
```
Comment (louis-bompart) [inline src/components/UserActions/ClickedDocumentList.ts]:
```suggestion
        super(element, ClickedDocumentList.ID, bindings) /* istanbul ignore next Istanbul issue with next*/;
```

Just so we got 💯 coverage (I know we don't need nor want to aim 💯 but we can, however, aim to that and improve the coverage by removing some erratic thing like this issue with Istanbul.

File: src/models/UserProfileModel.ts [hunk 2]
```diff
@@ -62,20 +69,18 @@ export class UserProfileModel extends Model {
...
+        Assert.isNotUndefined(this.options.organizationId);
+        Assert.isNotUndefined(this.options.accessToken);
 
         this.getOrFetchCache = {};
 
         this.endpoint = new UserProfilingEndpoint({
             uri: this.options.restUri,
-            accessToken: SEARCH_INTERFACE_ENDPOINT.accessToken,
-            organization: this.options.organization
+            accessToken: this.options.accessToken || (this.options.searchEndpoint as SearchEndpoint).accessToken,
```
Comment (louis-bompart) [inline src/models/UserProfileModel.ts]: Why the `as SearchEndpoint`, is accessToken not a prop of `ISearchEndpoint` ?

File: tests/utils.ts [hunk 3]
```diff
@@ -79,3 +61,8 @@ export function delay(callback: () => any, timemout = 0) {
         }, timemout);
     });
 }
+
+export function fakeUserProfileModel(root: HTMLElement, sandbox: SinonSandbox) {
```
Comment (louis-bompart) [inline tests/utils.ts]: Juicy comment for this ;)

File: src/models/UserProfileModel.ts (hunk 2 above)
Comment (jeremierobert-coveo) [reply src/models/UserProfileModel.ts]: Exactly only the class as accessToken

File: src/components/UserActions/ClickedDocumentList.ts (hunk 1 above)
Comment (jeremierobert-coveo) [reply src/components/UserActions/ClickedDocumentList.ts]: k

File: src/models/UserProfileModel.ts (hunk 2 above)
Comment (louis-bompart) [reply src/models/UserProfileModel.ts]: Next question, this.options.searchEndpoint is always a defined object?

Comment (jeremierobert-coveo) [reply src/models/UserProfileModel.ts]: Yes always

**Pull request merged**